dask/tests/test_cli.py

```diff
@@ -37,42 +38,105 @@ def test_config_get_bad_value():
 
 
 @pytest.fixture
-def tmp_conf_dir(tmpdir):
-    original = dask.config.PATH
-    dask.config.PATH = str(tmpdir)
+def tmp_conf_dir(tmpdir, monkeypatch):
+    # Set temporary DASK_CONFIG in dask.config and handle reloading of module
+    # before and after test so module initialization takes place setting attrs
+    # like PATH, config, paths, and module level constants
+    monkeypatch.setenv("DASK_CONFIG", str(tmpdir))
+    originals = {k: getattr(dask.config, k) for k in dir(dask.config)}
+    dask.config = importlib.reload(dask.config)
```

Comment:
This feels overkill? Is there a specific reason for it?

Reply:
I agree, but unfortunately it seems that way. Four other tests will fail in `test_config`. I wasn't immediately able to track down exactly why, but have a suspicion it's because a lot of function in the config module have parameters w/ defaults using globals like `config`, `defaults` etc. and for whatever reason it ends up effecting tests in the other test module. A bit tricky with globals that get set on import. Open to suggestions of course.